x-pack/legacy/plugins/maps/public/layers/styles/vector/properties/dynamic_style_property.d.ts

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+/* eslint-disable @typescript-eslint/consistent-type-definitions */
+
+import { IStyleProperty } from './style_property';
+import { FIELD_ORIGIN } from '../../../../../common/constants';
+import { FieldMetaOptions } from '../../../../../common/style_property_descriptor_types';
+import { IField } from '../../../fields/field';
+import { IVectorLayer } from '../../../vector_layer';
+import { IVectorSource } from '../../../sources/vector_source';
+import { CategoryFieldMeta, RangeFieldMeta } from '../../../../../common/descriptor_types';
+
+export interface IDynamicStyleProperty extends IStyleProperty {
+  getFieldMetaOptions(): FieldMetaOptions;
+  getField(): IField | undefined;
+  getFieldName(): string;
+  getFieldOrigin(): FIELD_ORIGIN | undefined;
+  getComputedFieldName(): string | undefined;
+  getFieldSource(): IVectorSource | undefined;
```

Comment:
not an issue with this PR, but seeing this interface layed out I think it highlights unnecessary redundancies.

imho: I'd remove from this interface:

- `getFieldSource`:  used here https://github.com/elastic/kibana/blob/e438234a25ba8c33613845532c88d4e4454a5e2d/x-pack/legacy/plugins/maps/public/layers/styles/vector/properties/dynamic_style_property.js#L35
- `getFieldName`: usages in `dynamic_[color|label|orientation|size]_form`
    - It also highlights an unnecessary double lookup here: https://github.com/elastic/kibana/blob/6e5e8c815e0eab44e78af025a8ddada46b6244e2/x-pack/legacy/plugins/maps/public/layers/styles/vector/components/color/dynamic_color_form.js#L62-L71
- `getComputedFieldName`: move it's implementation down into `dynamic_size_property`, as it's only used there.

Reply:
I moved `getFieldSource` to an internal method since its only used in DynamicStyleProperty. For the others, I created https://github.com/elastic/kibana/issues/59914 to keep this PR's scope in check.